### ISSUE TITLE
Mark - Community Utrecht - A11y getest

### ISSUE THREAD

**Issue by rianrietveld** [nl-design-system/candidate#201]
## Review toegankelijkheid voor de Mark component in Community bij Utrecht

- Te reviewen: [Mark in StoryBook Utrecht](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-mark--docs)
- Acceptatiecriteria: [NL Design System Acceptatiecriteria voor Mark](https://nldesignsystem.nl/mark)

### Hoe te testen

Ga naar de te reviewen component in StoryBook
- Bij de component, kies in de zijbalk de view Default
- Klik op het open nieuwe tab icon rechtsboven in de taakbalk.
- Je ziet nu alleen de component met de bijbehorende CSS.

## Review toegankelijkheid  door Rian op 26 november 2024

Gereviewd Mark component in StoryBook Utrecht
Let op: alleen de voorbeelden die `<mark>` gebruiken vallen binnen deze review, andere opties zoals het gebruik van enkel een `<span>` valideren sowieso niet.

- [Default](https://nl-design-system.github.io/utrecht/storybook-css/iframe.html?args=&id=css-mark--default&viewMode=story)
- [Alternate Markup](https://nl-design-system.github.io/utrecht/storybook-css/iframe.html?args=&id=css-mark--default&viewMode=story): Deze optie gebruikt een `<span>` en valt daarom buiten deze review.
- [Appearance in print](https://nl-design-system.github.io/utrecht/storybook-css/iframe.html?args=&id=css-mark--print&viewMode=story)
- [Windows High Contrast Mode](https://nl-design-system.github.io/utrecht/storybook/?path=/story/css_css-mark--windows-high-contrast)

HTML basisvoorbeeld:
`<mark class="utrecht-mark">The Quick Brown Fox Jumps Over The Lazy Dog</mark>`

#### Acceptatiecriteria

Samenvatting: de Mark component in Community bij Amsterdan voldoet niet aan de acceptatiecriteria voor toegankelijkheid van het NL Design System.
Fail voor Default en Windows High Contrast Mode.
- WCAG [1.4.11 Contrast van niet-tekstuele content](https://nldesignsystem.nl/wcag/1.4.11).
- WCAG [1.4.1 Gebruik van kleur](https://nldesignsystem.nl/wcag/1.4.1)

### Toegankelijkheid algemeen

- [x] Tekst in een andere taal heeft het juiste lang-attribuut
WCAG [3.1.2 Taal van onderdelen](https://nldesignsystem.nl/wcag/3.1.2)
Oordeel: Niet van toepassing, contextafhankelijk.

### Toegankelijkheid visueel ontwerp

- [ ] De gemarkeerde tekst is niet alleen herkenbaar aan de achtergrondkleur
WCAG [1.4.1 Gebruik van kleur](https://nldesignsystem.nl/wcag/1.4.1)
Oordeel: Voldoet alleen voor de optie "Appearce in Print", waar een outline is toegevoegd.

- [x] De gemarkeerde tekst heeft voldoende contrast met de achtergrondkleur
WCAG [1.4.3 Contrast (minimum)](https://nldesignsystem.nl/wcag/1.4.3)
Contrast Default en Appearance in print: #00000 tov #ffd633 = 14.90
Contrast Windows High Contrast Mode: #00000 tov #67b0ff = 9.23
Oordeel: Voldoet.

- [ ] Wanneer de achtergrondkleur van de Mark component anders is dan de default browserkleur, dan heeft deze een kleurcontrast van minimaal 3:1 met de achtergrondkleur van de pagina
WCAG [1.4.11 Contrast van niet-tekstuele content](https://nldesignsystem.nl/wcag/1.4.11)
Contrast Default: #FFFFFF tov #ffd633 = 1.40: voldoet niet
Contrast Appearance in print: #FFFFFF tov #00000 (outline)  = 21: voldoet
Contrast Windows High Contrast Mode: #FFFFFF tov #67b0ff = 2,27: voldoet niet
Oordeel:  Voldoet niet.

### Toegankelijkheid zoom en herschalen

- [x] Als je inzoomt naar 200% blijft alle tekst leesbaar
WCAG [1.4.4 Herschalen van tekst](https://nldesignsystem.nl/wcag/1.4.4)
Oordeel: Voldoet.

- [x] Als je de tekstafstand vergroot blijft de tekst in zijn geheel zichtbaar
WCAG [1.4.12 Tekstafstand](https://nldesignsystem.nl/wcag/1.4.12)
Oordeel: Voldoet

### Toegankelijkheid screenreader
- [x] Gebruik het juiste HTML-element voor de Mark component
WCAG [1.3.1 Info en relaties](https://nldesignsystem.nl/wcag/1.3.1)
Oordeel: Voldoet.

## Opmerkingen
- Voeg een extra visueel aspect toe om de gemarkeerde tekst heen, bijvoorbeeld een outline, naast alleen een achtegrondkleur, zoals bij de Apprearance in Print optie.
- De screenreader support van `<mark>` is wisselend. Adrian Roselli geeft wat oplossingen in: [Tweaking Text Level Styles](https://adrianroselli.com/2017/12/tweaking-text-level-styles.html).
